radiometric_normalization/gimage.py

```diff
@@ -0,0 +1,120 @@
+'''
+Copyright 2015 Planet Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import logging
+from collections import namedtuple
+
+import numpy
+from osgeo import gdal, gdal_array
+
+# In-memory geospatial image
+# Bands: list of uint16 numpy arrays, each holding a band of data
+# Alpha: uint16 numpy array holding the alpha information
+# Metadata: dict containing georeferencing information
+# (geotransform, projection, rpc)
+GImage = namedtuple('GImage', 'bands, alpha, metadata')
+
+
+def save(gimage, filename):
+
+    band_count = len(gimage.bands) + 1
+    options = ['ALPHA=YES']
+
+    if band_count == 4:
+        options.append('PHOTOMETRIC=RGB')
+
+    datatype = gdal_array.NumericTypeCodeToGDALTypeCode(
+        gimage.bands[0].dtype.type)
+    ysize, xsize = gimage.bands[0].shape
+    gdal_ds = gdal.GetDriverByName('GTIFF').Create(
+        filename, xsize, ysize, band_count, datatype,
+        options=options)
+
+    for i in range(len(gimage.bands)):
```

Comment:
Consider using enumerate:

    for i, band in enumerate(gimage.bands):
        gdal_array.BandWriteArray(gdal_ds.GetRasterBand(i + 1), band)